src/jit/gentree.cpp

```diff
@@ -15469,34 +15477,39 @@ bool GenTree::canBeContained() const
 //    Essentially, it will be rolled into the code generation for the parent.
 //
 // Assumptions:
-//    This method relies upon the value of gtRegNum field to determine whether the tree node
-//    is contained.
-//    Therefore you can not call this method until after the LSRA phase has allocated physical
-//    registers to the treenodes.
+//    This method relies upon the value of the GTF_CONTAINED flag.
+//    Therefore this method is only valid after Lowering.
+//    Also note that register allocation or other subsequent phases may cause
+//    nodes to become contained (or not) and therefore this property may change.
 //
 bool GenTree::isContained() const
 {
+#ifdef LEGACY_BACKEND
+    return false;
+#else // !LEGACY_BACKEND
     assert(IsLIR());
+    bool returnValue = ((gtFlags & GTF_CONTAINED) != 0);
 
+#ifdef DEBUG
     if (!canBeContained() || gtHasReg())
     {
-        return false;
+        assert(returnValue == false);
```

Comment:
My preference would be to leave this as-is. Otherwise, I would probably change the name of the variable to something like `isMarkedContained` - otherwise `!returnValue` seems to imply "there is no return value" rather than "the return value is false".